src/main/java/nu/studer/gradle/rocker/RockerCompile.java

```diff
@@ -160,50 +154,48 @@ public void execute(CopySpec spec) {
                 });
 
                 // generate the files from the modified templates
-                execResult = executeRocker(tempDir);
+                executeRocker(tempDir);
             }
         }
 
         // for the Gradle Build Cache to function properly, the same inputs must always create exactly the same output
         // thus, if hot-reloading is disabled and the generated source code contains timestamps, we remove the MODIFIED_AT line
         RockerVersion rockerVersion = RockerVersion.fromProject(getProject());
         if (config.isOptimize() && rockerVersion.generatesRedundantCode_MODIFIED_AT()) {
+            getWorkerExecutor().await();
```

Comment:
Note that this will prevent the task action to release locks and to be run in parallel.
What you could do until work item chaining is possible is to put this cleanup work in another task action (`doLast`).